### ISSUE TITLE
CoilCoolingDXSingleSpeed requires an OptionalDouble for setRatedCOP

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio#2054]
```
clg_coil = OpenStudio::Model::CoilCoolingDXSingleSpeed.new(model)

 [1] (main)> clg_coil.setRatedCOP(4)
TypeError: Expected argument 1 of type boost::optional< double >, but got Fixnum 4
    in SWIG method 'setRatedCOP'
from (pry):209:in `setRatedCOP'
```

This works:

```
[2] Turkevi(main)> clg_coil.setRatedCOP(OpenStudio::OptionalDouble.new(4))
=> nil
```

**Comment by MatthewSteen** [comment]
https://unmethours.slack.com/archives/openstudio-measures/p1452890055000002

https://github.com/NREL/OpenStudio/issues/382

**Comment by DavidGoldwasser** [comment]
@kbenne if we fix this I assume we need to handle measures that are already written to pass in an optional double, but then also accept a double.

**Comment by shorowit** [comment]
Subscribing.

**Comment by jmarrec** [comment]
There are plenty of other objects and methods that have this problem, eg:
[CoilCoolingDXTwoSpeed::setRatedHighSpeedTotalCoolingCapacity](
https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/CoilCoolingDXTwoSpeed.hpp#L281)

**Comment by jmarrec** [comment]
```sh
cd OpenStudio/openstudiocore/src/model/ && grep -r --include \*[^l].hpp "set.*boost::optional" ./
```

**Comment by jmarrec** [comment]
Here's the list without the false positives:

```
CoilCoolingDXSingleSpeed.hpp::RatedCOP
CoilCoolingDXSingleSpeed.hpp::RatedEvaporatorFanPowerPerVolumeFlowRate
CoilCoolingDXSingleSpeed.hpp::NominalTimeForCondensateRemovalToBegin
CoilCoolingDXSingleSpeed.hpp::RatioOfInitialMoistureEvaporationRateAndSteadyStateLatentCapacity
CoilCoolingDXSingleSpeed.hpp::MaximumCyclingRate
CoilCoolingDXSingleSpeed.hpp::LatentCapacityTimeConstant
CoilCoolingDXSingleSpeed.hpp::CondenserAirInletNodeName
CoilCoolingDXSingleSpeed.hpp::EvaporativeCondenserEffectiveness
CoilCoolingDXSingleSpeed.hpp::EvaporativeCondenserAirFlowRate
CoilCoolingDXSingleSpeed.hpp::EvaporativeCondenserPumpRatedPowerConsumption
CoilCoolingDXSingleSpeed.hpp::CrankcaseHeaterCapacity
CoilCoolingDXSingleSpeed.hpp::MaximumOutdoorDryBulbTemperatureForCrankcaseHeaterOperation
CoilCoolingDXSingleSpeed.hpp::BasinHeaterCapacity
CoilCoolingDXSingleSpeed.hpp::BasinHeaterSetpointTemperature
CoilCoolingDXTwoSpeed.hpp::RatedHighSpeedTotalCoolingCapacity
CoilCoolingDXTwoSpeed.hpp::RatedHighSpeedSensibleHeatRatio
CoilCoolingDXTwoSpeed.hpp::RatedHighSpeedAirFlowRate
CoilCoolingDXTwoSpeed.hpp::RatedLowSpeedTotalCoolingCapacity
CoilCoolingDXTwoSpeed.hpp::RatedLowSpeedSensibleHeatRatio
CoilCoolingDXTwoSpeed.hpp::RatedLowSpeedAirFlowRate
CoilCoolingDXTwoSpeed.hpp::HighSpeedEvaporativeCondenserAirFlowRate
CoilCoolingDXTwoSpeed.hpp::HighSpeedEvaporativeCondenserPumpRatedPowerConsumption
CoilCoolingDXTwoSpeed.hpp::LowSpeedEvaporativeCondenserAirFlowRate
CoilCoolingDXTwoSpeed.hpp::LowSpeedEvaporativeCondenserPumpRatedPowerConsumption
CoilHeatingWaterToAirHeatPumpEquationFit.hpp::RatedAirFlowRate
CoilHeatingWaterToAirHeatPumpEquationFit.hpp::RatedWaterFlowRate
CoilHeatingWaterToAirHeatPumpEquationFit.hpp::RatedHeatingCapacity
Relationship.hpp::RelatedModelObject
ZoneHVACWaterToAirHeatPump.hpp::SupplyAirFlowRateDuringCoolingOperation
ZoneHVACWaterToAirHeatPump.hpp::SupplyAirFlowRateDuringHeatingOperation
ZoneHVACWaterToAirHeatPump.hpp::SupplyAirFlowRateWhenNoCoolingorHeatingisNeeded
ZoneHVACWaterToAirHeatPump.hpp::OutdoorAirFlowRateDuringCoolingOperation
ZoneHVACWaterToAirHeatPump.hpp::OutdoorAirFlowRateDuringHeatingOperation
ZoneHVACWaterToAirHeatPump.hpp::OutdoorAirFlowRateWhenNoCoolingorHeatingisNeeded
ZoneHVACWaterToAirHeatPump.hpp::MaximumCyclingRate
ZoneHVACWaterToAirHeatPump.hpp::HeatPumpTimeConstant
ZoneHVACWaterToAirHeatPump.hpp::FractionofOnCyclePowerUse
ZoneHVACWaterToAirHeatPump.hpp::HeatPumpFanDelayTime
ZoneHVACWaterToAirHeatPump.hpp::MaximumSupplyAirTemperaturefromSupplementalHeater
ZoneHVACWaterToAirHeatPump.hpp::MaximumOutdoorDryBulbTemperatureforSupplementalHeaterOperation
```

**Comment by jmarrec** [comment]
Fixed for all above objects in #2589.

**Comment by jmarrec** [comment]
I forgot to close this issue when in fixed it in #2589
